internal/model/db.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2021 The golang.design Initiative Authors.
+// All rights reserved. Use of this source code is governed
+// by a MIT license that can be found in the LICENSE file.
+//
+// Originally written by Changkun Ou <changkun.de> at
+// changkun.de/s/redir, adopted by Mai Yang <maiyang.me>.
+
+package model
+
+import (
+	"context"
+	"errors"
+	"time"
+)
+
+var (
+	ErrExistedAlias = errors.New("alias is existed")
+)
+
+type AliasKind int
+
+const (
+	KindShort AliasKind = iota
+	KindRandom
+)
+
+const (
+	dbname = "redir"
+)
+
+// Redirect records a kind of alias and its correlated link.
+type Redirect struct {
+	Alias   string    `json:"alias"   db:"alias"`
+	Kind    AliasKind `json:"kind"    db:"kind"`
+	URL     string    `json:"url"     db:"url"`
+	Private bool      `json:"private" db:"private"`
+}
+
+// Visit indicates an Record of Visit pattern.
+type Visit struct {
+	Alias   string    `json:"alias"   db:"alias"`
+	Kind    AliasKind `json:"kind"    db:"kind"`
+	IP      string    `json:"ip"      db:"ip"`
+	UA      string    `json:"ua"      db:"ua"`
+	Referer string    `json:"referer" db:"referer"`
+	Time    time.Time `json:"time"    db:"time"`
+}
+
+type Refstat struct {
+	Referer string `json:"referer"`
```

Comment:
```suggestion
	Referrer string `json:"referrer"`
```

好像所有的 referrer 都 typo 了，得统一改一下

Reply:
https://developer.mozilla.org/zh-CN/docs/Web/HTTP/Headers/Referer

Reply:
改不改都可以，看看要不要改